web/exercises/replyGraphGelly.md

```diff
@@ -0,0 +1,103 @@
+---
+layout: page
+title: Gelly API - Reply Graph with Page Rank
+permalink: /exercises/replyGraphGelly.html
+---
+The task of the “Reply Graph with Page Rank” exercise is to extract reply connections between mails in the archives of Apache Flink’s developer mailing list, define a reply graph from these connections and to analyze the structure of the Flink community by running the PageRankAlgorithm over the generated graph.
+
+A reply connection is defined as a pair of two email addresses (`Tuple2<String, String>`) where the first email address replied to an email of the second email address. The task of this exercise is to first compute all reply connections between emails in the Mail Data Set and count the number of reply connections for each pair of users. However, in counting the connections, duplicate emails need to be taken into consideration i.e. some users can reply from different email addresses. The results obtained need to be used to generate a graph structure which can be given as input to the PageRankAlgorithm.
+
+### Input Data
+
+This exercise uses the Mail Data Set which was extracted from the Apache Flink development mailing list archive. The [Mail Data Set instructions]({{ site.baseurl }}/exercises/mailData.html) show how to read the data set in a Flink program using the `CsvInputFormat`.
+
+The task requires three fields `MessageID`, `Sender`, and `Reply-To`. The input data can be read as a `DataSet<Tuple3<String, String, String>>`. When printed, the data set should look similar to this:
+
+~~~
+(<CAAdrtT0-sfxxUK-BrPC03ia7t1WR_ogA5uA6J5CSRvuON+snTg@mail.gmail.com>,Fabian Hueske <fhueske@apache.org>,<C869A196-EB43-4109-B81C-23FE9F726AC6@apache.org>)
+(<CANMXwW0HOvk7n=h_rTv3RbK0E4ti1D7OdsY_3r8joib6rAAt2g@mail.gmail.com>,Aljoscha Krettek <aljoscha@apache.org>,<CANC1h_vn8E8TLXD=8szDN+0HO6JrU4AsCWgrXh8ojkA=FiPxNw@mail.gmail.com>)
+(<0E10813D-5ED0-421F-9880-17C958A41724@fu-berlin.de>,Ufuk Celebi <u.celebi@fu-berlin.de>,null)
+~~~
+
+**Note**, the `Reply-To` field might have the value `"null"` indicating that this mail was not written in repsonse to another mail.
+
+### Expected Output
+
+
+The result for the first part of the exercise should be a `DataSet<Tuple3<String, String, Integer>>`. The first field is the sender email address of the reply mail, the second field is the sender email address of the mail that was replied to, and the third field is the number of reply connections between these two email addresses. When printed, the data set should look like this:
+
+~~~
+(fhueske@gmail.com,hongbosoftware@163.com,1)
```

Comment:
Please use only email addresses of Flink committers (preferable only dataArtisans employees) for the training material. We don't want somebody else to be surprised to find their address here.